### ISSUE TITLE
[bitnami/ghost] startup issues when upgrading past v5.68.0

### ISSUE THREAD

**Issue by 97WaterPolo** [bitnami/containers#72308]
### Name and Version

bitnami/ghost:5.68.0

### What architecture are you using?

amd64

### What steps will reproduce the bug?

1. Using Unraid (or docker command line), pull and run the containers v5.68.0
2. Application will start and run, 
![image](https://github.com/user-attachments/assets/62aae725-a8a8-47f6-964a-67a8ed221bbb)

3. Using Unraid (or docker command line), upgrade and pull the next version v5.69.0
4. Application has the same startup message but the container has exited
![image](https://github.com/user-attachments/assets/c6dc0dc3-4e8a-420d-b361-5d2f860bf0b3)

### What is the expected behavior?

I expected the upgrade to complete without any issues, I was on a 5.3.X and jumped to latest and it failed to start. I slowly tried differing versions until I came from the upgrade to 5.68.0 > 5.69.0 which is when it started to fail to start. I don't see any logs in the docker container logs and the container isn't ever on long enough for me to check the logs while it is running.

### What do you see instead?

The container fails to start once upgrading to 5.69.0

### Additional information
```
Docker Client:
 Version:           20.10.21
 API version:       1.41
 Go version:        go1.18.7
 Git commit:        baeda1f
 Built:             Tue Oct 25 17:56:30 2022
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.21
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.18.7
  Git commit:       3056208
  Built:            Tue Oct 25 18:02:03 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.6.9
  GitCommit:        1c90a442489720eec95342e1789ee8a5e1b9536f
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d1
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Comment by 97WaterPolo** [comment]
![image](https://github.com/user-attachments/assets/efc699fe-b9ce-4131-a5a9-6c744e4a5f20)

Looks like the only difference I see in the console output is the Executing SQL Command. There isn't any other debug showing up

**Comment by javsalgar** [comment]
Another option to further debug the issue could be running a sleep infinity as entrypoint in the ghost container, then enter the container and run this command:

```
/opt/bitnami/scripts/ghost/entrypoint.sh /opt/bitnami/scripts/ghost/run.sh
```

If that fails, you can also run the command `ghost run` with NODE_ENV=development 

https://forum.ghost.org/t/how-to-enable-debugging-output-in-ghost/8479/5

**Comment by 97WaterPolo** [comment]
This is after running the command on a sleep infinity version of it!

Since I am trying to debug it, I then run ghost run only for it to exit the running process as soon as I enter the command, just like when I run the script.

![image](https://github.com/user-attachments/assets/237446e6-e4ed-4c42-9c27-c08510f0d45e)

On the other hand if I built it from the 5.68.0 version that does work, when I execute your entrypoint.sh and run.sh it doesn't return, which makes me thing the process is running. Not sure why it is breaking out of the process once I upgrade past that version.

![image](https://github.com/user-attachments/assets/6fc9bfe8-501f-4795-bf9a-3463e33ddb2b)

**Comment by 97WaterPolo** [comment]
I did some more digging, I run ghost with the -D! parameter and I got an error, looks like in this version it fails to open SQLITE database? Which is weird as everything I have is configured to use mysql. Any idea why it would be trying to initialize SqlLite?

![image](https://github.com/user-attachments/assets/95f759cd-2a61-4646-89ae-dca6d37ed471)

**Comment by javsalgar** [comment]
Hi,

Thank you so much for all the digging. This behavior of defaulting to sqlite when the config returns that the database should be mysql seems more like an upstream issue rather than an issue with the Bitnami packaging. I think it is worth sharing with the upstream developers.

**Comment by 97WaterPolo** [comment]
Unfortunately they are rejecting it as a bug and stating that it is an environment issue. Going through more of the Ghost documentation it looks like the ghost configuration is loaded from a config.json, not environment variables that I have provided for this container. Any chance you know where those config files are stored within the container and I can manually validate / alter it to see if I can get it working? 

Do you think it has anything to do with the env vars? It looks like the old template I used for UnraidOS has it mapped to MARIADB_HOST and MARIADB_PORT_NUMBER rather than GHOST_DATABASE_HOST and GHOST_DATABASE_PORT_NUMBER? If the env vars changed then my legacy container 5.68.X might use those two env vars but when upgrading past 5.69.X it references the new ones prefixed with GHOST_*. Do you think this could be a possibility that if it doesn't find the database host it fallbacks to Sqlite3?

![image](https://github.com/user-attachments/assets/6642ef02-3c82-4a2d-ae89-33155a7829ca)

**Comment by javsalgar** [comment]
Hi!

The configuration file should be in /opt/bitnami/ghost/config.production.json

**Comment by 97WaterPolo** [comment]
![image](https://github.com/user-attachments/assets/a59c3369-a423-43e2-88c5-348061642eae)

I ended up finding it in another location but for some reason it looks like it is not pulling the correct values from ENV variables

**Comment by 97WaterPolo** [comment]
v5.94.0
![image](https://github.com/user-attachments/assets/d26134c4-29ee-477f-b8aa-808cbd9a903e)

v5.68.0
![image](https://github.com/user-attachments/assets/e8bb36c5-f592-4304-affe-83b9b10242b2)

Looks like the /opt/bitnami/ghost/versions/5.XX.0/node_modules/knex-migrator/lib/database.js is getting the incorrect client, any support from Ghost is claiming it is a docker environment issue not an issue with the application.

**Comment by 97WaterPolo** [comment]
![image](https://github.com/user-attachments/assets/f9de4cc6-e5e4-4a06-a549-a6de4f2bf2eb)

I was running a previous version of mariadb that doesn't seem to support the ALTER TABLE command
`/usr/bin/mysql  Ver 15.1 Distrib 10.4.21-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2`

I ended up upgrading to 
`/usr/bin/mysql  Ver 15.1 Distrib 10.11.8-MariaDB, for Linux (x86_64) using readline 5.1`

which is now causing this error, the syntax support should be there, not sure why it is still failing the database migration when I run `ghost log`
![image](https://github.com/user-attachments/assets/935c1faf-c7ff-4d37-9a3b-2acfe9b35b2a)


Any suggestions on what I should do?

**Comment by javsalgar** [comment]
Are you able to run that alter table command manually? Just to ensure if the syntax is supported

**Comment by 97WaterPolo** [comment]
I tried updating to 5.69.0 and it still failed with the same error.

I created an empty table with a column matching the statement, I tried executing it and it failed
![image](https://github.com/user-attachments/assets/6831c0ee-4b32-4111-93bb-d9ea7d981901)

I then did a SELECT VERSION() and tried it again and without a restart or any other changes it worked.
![image](https://github.com/user-attachments/assets/b07b5770-202d-472d-82e9-15f500e49b69)

I restarted with version 5.69.0 and it looked like it migrated correctly, from this point on I was able to go to the latest and got everything going. Not sure why I had issues after I upgraded the database, but it seems to be all resolved now. Thanks for all the help troubleshooting!
